src/riak_kv_w1c_worker.erl

```diff
@@ -108,13 +108,13 @@ put(RObj0, Options) ->
                 PW :: pos_integer(),
                 Bucket :: binary()|{binary(), binary()},
                 NVal :: pos_integer(),
-                Key :: binary()|{binary(), binary()},
+                Key :: tuple()|{binary(), tuple()},
                 EncodeFn :: fun((riak_object:riak_object()) -> binary()),
                 Preflist :: term()) ->
                        {ok, {reference(), atom()}}.
 
-async_put(RObj, W, PW, Bucket, NVal, {PK, LK}, EncodeFn, Preflist) when is_tuple(LK) ->
-    async_put(RObj, W, PW, Bucket, NVal, PK, LK, EncodeFn, Preflist);
+async_put(RObj, W, PW, Bucket, NVal, {PartIdx, LK}, EncodeFn, Preflist) when is_tuple(LK) ->
```

Comment:
Something feels off about this. I can see in `riak_kv_ts_api:put_data_to_partitions` where you put in the change to pass in `{PartIdx, LK}`, but then the other place that `async_put` is called (in `riak_kv_w1c_worker:put`) it looks like it's still passing in something else. I'm not quite sure if/what the mismatch here is stemming from, or if there's a bug being introduced somewhere, so I will keep investigating and try to figure out what's going on here.